src/test/java/SearchServletTest.java

```diff
@@ -88,43 +88,116 @@ public void tearDown() {
 
   @Test
   public void queryWithAllFieldsFilled() throws IOException {
+    // Columns ommitted from database visual: id, userId, blobKey, imageUrl, rawText.
+    //
+    // id   Timestamp      Price          Store                    Categories
+    // 1  1045237591000    26.12        "walmart"         ["candy", "drink", "personal"]
+    // 2  1560193140000    14.51        "contoso"         ["cappuccino", "sandwich", "lunch"]
+    // 3  1491582960000    29.01   "main st restaurant"   ["food", "meal", "lunch"]
+    //
+    // Query: drink, 2/1/03-2/28/03, walmart, $5.00-$30.00.
+    // Will only return walmart receipt.
+
     // Add mock receipts to datastore.
     TestUtils.addTestReceipts(datastore);
 
     // Perform doGet - this should retrieve one receipt.
-    TestUtils.setRequestParameters(
-        request, CST_TIMEZONE_ID, CATEGORIES, SHORT_DATE_RANGE, STORE, MIN_PRICE, MAX_PRICE);
+    TestUtils.setSearchServletRequestParameters(
+        request, CST_TIMEZONE_ID, CATEGORY, SHORT_DATE_RANGE, STORE, MIN_PRICE, MAX_PRICE);
+    servlet.doGet(request, response);
+    writer.flush();
+
+    // Make sure receipt is retrieved by finding the receipt id in the writer.
+    Assert.assertTrue(stringWriter.toString().contains("\"id\":1"));
+  }
+
+  @Test
+  public void queryWithoutStore() throws IOException {
+    // Columns ommitted from database visual: id, userId, blobKey, imageUrl, rawText.
+    //
+    // id   Timestamp      Price          Store                    Categories
+    // 1  1045237591000    26.12        "walmart"         ["candy", "drink", "personal"]
+    // 2  1560193140000    14.51        "contoso"         ["cappuccino", "sandwich", "lunch"]
+    // 3  1491582960000    29.01   "main st restaurant"   ["food", "meal", "lunch"]
+    //
+    // Query: drink, 2/1/03-2/28/03, $5.00-$30.00.
+    // Will return main st restaurant and contoso receipts.
+
+    // Add mock receipts to datastore.
+    TestUtils.addTestReceipts(datastore);
+
+    // Perform doGet - this should retrieve a couple receipts.
+    TestUtils.setSearchServletRequestParameters(
+        request, CST_TIMEZONE_ID, CATEGORY, SHORT_DATE_RANGE, EMPTY_STRING, MIN_PRICE, MAX_PRICE);
     servlet.doGet(request, response);
     writer.flush();
 
-    // Make sure receipt is retrieved by finding the store name in the writer.
-    Assert.assertTrue(stringWriter.toString().contains("walmart"));
+    // Make sure receipts are retrieved by finding receipt ids in the writer.
+    Assert.assertTrue(stringWriter.toString().contains("\"id\":1"));
```

Comment:
Can you assert on the other receipts that should be returned? Same for the others.